### PR TITLE
Added NumbaLinker

### PR DESCRIPTION
**Pull request by kc611**
Adds a `NumbaLinker` based upon the new `JitLinker` class.

Based upon machinery from https://github.com/pymc-devs/aesara/pull/365 

Cc @brandonwillard

### CODE REVIEW

**Review by brandonwillard — COMMENTED** [review]
Well, it looks like you did it!  That's about all it should take.

File: aesara/link/numba/dispatch.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+def numba_funcify_ScalarAdd(op, **kwargs):
+    import numpy as np
+
+    numpy_func = getattr(np, op.nfunc_spec[0])
+
+    @numba.njit
+    def scalar_func(*args):
+        result = args[0]
+        for arg in args[1:]:
+            result = numpy_func(arg, result)
```
Comment (brandonwillard) [inline aesara/link/numba/dispatch.py:60]: Is this an instance in which the Numba conversion was having trouble dealing with varargs (e.g. `numpy_func(*args)`)?

Comment (kc611) [reply aesara/link/numba/dispatch.py:60]: That error occurs when we put a `numba.vectorize` tag on a function which (or it's child functions) has varargs like `*args` . It's the `TODO` listed in `Elemwise` 's Numba conversion.

**Comment by kc611** [comment]
Continued in https://github.com/pymc-devs/aesara/pull/372

**Pull request closed without merging**